File: matchups.php
<?php
function winloss($results) {
	$wins = 0;
	$losses = 0;
	if (isset($results['win'])) {
		$wins = $results['win'];
	} else {
		$wins = 0;
	}

	if (isset($results['loss'])) {
		$losses = $results['loss'];
	} else {
		$losses = 0;
	}

	return "$wins - $losses";
}

function clean_name($name) {
	$name = strtolower($name);
	$name = ucfirst($name);
	return $name;
}

$file = file_get_contents('player_matchups.json');
$json = json_decode($file, true);

if (isset($_GET['name'])) {
	$player = $_GET['name'];
} else {
	print "<title>Players</title>";
	print "<b>Player</b><br>";
	ksort($json);
	foreach ($json as $name => $val) {
		print "<a href='matchups.php?name=$name'>$name</a><br>";
	}
	exit();
}

$player = clean_name($player);

if (!isset($json[$player])) {
	print "<b>Player not Found</b>";
	exit();
}

$matchups = $json[$player];

print "<title>$player</title>";

print "<b>$player</b> (". round($json[$player]['win_pct'], 2) ."%)";
print '<br>';

ksort($matchups);
foreach ($matchups as $key => $val) {
	if ($key == "win_pct") {
		continue;
	}
	print "<a href='matchups.php?name=$key'>$key</a>: ".winloss($val);
	print '<br>';
}

?>

File: matchups.py
#!/usr/bin/env python
import cgi
import json
from mako.template import Template

print 'Content Type: "text/html"'
print

html = '''
<html>
<head>
<title>
% if player:
    ${player}
% else:
    Matchups
% endif
</title>
</head>
<body>
% if error:
    <b style='color: red'>${error}</b>
% elif player is None:
    <b>Players</b>
    <br>
    <ul>
    % for p in sorted(players):
        <li>${player_link(p)}</li>
    % endfor
    </ul>

% else:
    <b>${player} (${'{:.2f}'.format(win_pct)}%)</b>
    <br>

    <ul>
    % for opp, wl in opps:
        <li>${player_link(opp)}: ${wl}</li>
    % endfor
    </ul>
% endif

<%def name="player_link(p)">
<a href="matchups.py?name=${p}">${p}</a>
</%def>

</body>
</html>
'''

def winloss(results):
    wins = results.get('win', 0)
    losses = results.get('loss', 0)

    return '{} - {}'.format(wins, losses)

with open('player_matchups.json') as j:
    players = json.load(j)

form = cgi.FieldStorage()
player = form.getvalue('name')

if player is None:
    print Template(html).render(player=None,
                                players=players)
    exit()
else:
    player = player.title()

if player not in players:
    error = 'Player not found'

    print Template(html).render(error=error)
    exit()

error = None

matchups = players[player]
win_pct = matchups['win_pct']

opps = []
for opp, val in sorted(matchups.items()):
    if opp == 'win_pct':
        continue

    opps.append((opp, winloss(val)))

print Template(html).render(error=error,
                            player=player,
                            win_pct=win_pct,
                            opps=opps)
